src/sentry/static/sentry/app/components/events/interfaces/spans/minimap.tsx

```diff
@@ -0,0 +1,613 @@
+    const traceViewDOM = this.props.traceViewRef.current;
+
+    if (!canvas || !traceViewDOM) {
+      return;
+    }
+
+    const canvasContext = canvas.getContext('2d');
+
+    if (!canvasContext) {
+      return;
+    }
+
+    const rootRect = rectOfContent(traceViewDOM);
+
+    const scaleX = canvas.clientWidth / rootRect.width;
+    const scaleY = canvas.clientHeight / rootRect.height;
+
+    // https://www.html5rocks.com/en/tutorials/canvas/hidpi/
+    // we consider the devicePixelRatio (dpr) factor so that the canvas looks decent on hidpi screens
+    // such as retina macbooks
+    const devicePixelRatio = window.devicePixelRatio || 1;
+
+    const resizeCanvas = (width: number, height: number) => {
+      // scale the canvas up by the dpr factor
+      canvas.width = width * devicePixelRatio;
+      canvas.height = height * devicePixelRatio;
+
+      // scale the canvas down by the dpr factor thru CSS
+      canvas.style.width = '100%';
+      canvas.style.height = `${height}px`;
+    };
+
+    resizeCanvas(rootRect.width * scaleX, rootRect.height * scaleY);
+
+    canvasContext.setTransform(1, 0, 0, 1, 0, 0);
+    canvasContext.clearRect(0, 0, canvas.width, canvas.height);
+    canvasContext.scale(scaleX, scaleY);
+
+    // scale canvas operations by the dpr factor
+    canvasContext.scale(devicePixelRatio, devicePixelRatio);
+
+    const black = (pc: number) => `rgba(0,0,0,${pc / 100})`;
+    const back = black(0);
+
+    const drawRect = (
+      rect: {x: number; y: number; width: number; height: number},
+      colour: string
+    ) => {
+      if (colour) {
+        canvasContext.beginPath();
+        canvasContext.rect(rect.x, rect.y, rect.width, rect.height);
+        canvasContext.fillStyle = colour;
+        canvasContext.fill();
+      }
+    };
+
+    // draw background
+
+    drawRect(rectRelativeTo(rootRect, rootRect), back);
+
+    // draw the spans
+
+    Array.from(traceViewDOM.querySelectorAll<HTMLElement>('[data-span="true"]')).forEach(
+      el => {
+        const backgroundColor = window.getComputedStyle(el).backgroundColor || black(10);
+        drawRect(rectRelativeTo(rectOfElement(el), rootRect), backgroundColor);
+      }
+    );
+  };
+
+  renderMinimapCursorGuide = () => {
+    if (!this.state.showCursorGuide || !this.state.mousePageX) {
+      return null;
+    }
+
+    const minimapCanvas = this.props.minimapInteractiveRef.current;
+
+    if (!minimapCanvas) {
+      return null;
+    }
+
+    const rect = rectOfContent(minimapCanvas);
+
+    // clamp mouseLeft to be within [0, 100]
+    const mouseLeft = clamp(
+      ((this.state.mousePageX - rect.x) / rect.width) * 100,
+      0,
+      100
+    );
+
+    return (
+      <line
+        x1={`${mouseLeft}%`}
+        x2={`${mouseLeft}%`}
+        y1="0"
+        y2={MINIMAP_HEIGHT}
+        strokeWidth="1"
+        strokeOpacity="0.7"
+        style={{stroke: '#E03E2F'}}
+      />
+    );
+  };
+
+  renderViewHandles = ({
```

Comment:
It might make Minimap a bit easier to read if some of these render methods were components.